vnext/Microsoft.ReactNative/Views/TextInputViewManager.cpp

```diff
@@ -267,51 +260,49 @@ void TextInputShadowNode::registerEvents() {
       eventDataEndEditing = folly::dynamic::object("target", tag)(
           "text", HstringToDynamic(control.as<xaml::Controls::PasswordBox>().Password()));
     }
-    if (!m_updating && instance != nullptr) {
-      instance->DispatchEvent(tag, "topTextInputBlur", std::move(eventDataBlur));
-      instance->DispatchEvent(tag, "topTextInputEndEditing", std::move(eventDataEndEditing));
+    if (!m_updating) {
+      GetViewManager()->GetReactContext().DispatchEvent(tag, "topTextInputBlur", std::move(eventDataBlur));
+      GetViewManager()->GetReactContext().DispatchEvent(tag, "topTextInputEndEditing", std::move(eventDataEndEditing));
     }
   });
 
   m_controlKeyDownRevoker =
       control.KeyDown(winrt::auto_revoke, [=](auto &&, xaml::Input::KeyRoutedEventArgs const &args) {
         if (args.Key() == winrt::Windows::System::VirtualKey::Enter && !args.Handled()) {
-          if (auto instance = wkinstance.lock()) {
-            folly::dynamic eventDataSubmitEditing = {};
-            if (m_isTextBox) {
-              eventDataSubmitEditing = folly::dynamic::object("target", tag)(
-                  "text", HstringToDynamic(control.as<xaml::Controls::TextBox>().Text()));
-            } else {
-              eventDataSubmitEditing = folly::dynamic::object("target", tag)(
-                  "text", HstringToDynamic(control.as<xaml::Controls::PasswordBox>().Password()));
-            }
-            instance->DispatchEvent(tag, "topTextInputSubmitEditing", std::move(eventDataSubmitEditing));
+          folly::dynamic eventDataSubmitEditing = {};
+          if (m_isTextBox) {
+            eventDataSubmitEditing = folly::dynamic::object("target", tag)(
+                "text", HstringToDynamic(control.as<xaml::Controls::TextBox>().Text()));
+          } else {
+            eventDataSubmitEditing = folly::dynamic::object("target", tag)(
+                "text", HstringToDynamic(control.as<xaml::Controls::PasswordBox>().Password()));
           }
```

Comment:
nit: Should we factor this a bit cleaner, both branches only differ by control type, yet share a lot of code together.

```suggestion
          auto text = m_isTextBox 
              ? control.as<xaml::Controls::TextBox>().Text()
              : control.as<xaml::Controls::PasswordBox>().Password();
          eventDataSubmitEditing = folly::dynamic::object("target", tag)(
              "text", HstringToDynamic(text));
```